src/Interface.js

```diff
@@ -0,0 +1,248 @@
+import * as Constant from './Constant.js'
+
+export function prepare (target, registry) {
+	target.textContent = ''
+}
+
+export function create (element, xmlns, from) {
+	if (from === Constant.search) {
+		return search(element, index) || create(element, xmlns, -from)
+	} else {
+		switch (element.uuid) {
+			case Constant.node:
+				if (xmlns) {
+					return document.createElementNS(xmlns, element.type)
+				} else {
+					return document.createElement(element.type)
+				}
+			case Constant.text:
+			case Constant.empty:
+				return document.createTextNode(element.children)
+			case Constant.comment:
+				return document.createComment(element.children)
+			case Constant.portal:
+				return target(element.type)
+			case Constant.fragment:
+				return document.createDocumentFragment()
+		}
+	}
+
+	return element.owner
+}
+
+export function remove (parent, element) {
+	parent.owner.removeChild(element.owner)
+}
+
+export function append (parent, element) {
+	parent.owner.appendChild(element.owner)
+}
+
+export function insert (parent, element, sibling) {
+	parent.owner.insertBefore(element.owner, sibling.owner)
+}
+
+export function content (element, value) {
+	element.owner.nodeValue = value
+}
+
+export function valid (value, type) {
+	switch (type) {
+		case Constant.event:
+			return value.BUBBLING_PHASE === 3
+		case Constant.owner:
+		case Constant.component:
+			return value.ELEMENT_NODE === 1
+	}
+}
+
+export function owner (value) {
+	return value
+}
+
+export function target (value) {
+	if (value) {
+		return typeof value === 'string' ? document.querySelector(value) : value
+	} else {
+		return document.documentElement
+	}
+}
+
+export function namespace (element, xmlns) {
+	switch (element.type) {
+		case 'svg':
+			return 'http://www.w3.org/2000/svg'
+		case 'math':
+			return 'http://www.w3.org/1998/Math/MathML'
+		case 'foreignObject':
+			return ''
+		default:
+			return xmlns
+	}
+}
+
+export function event (element, event) {
+	return element.state[event.type]
+}
+
+export function props (element, props, from) {
+	switch (from) {
+		case Constant.update:
+			switch (element.type) {
+				case 'input':
+					return merge({type: null, step: null, min: null, max: null}, props)
+				case 'select':
+					if (props.defaultValue != null || props.multiple) {
+						return Utility.assign({value: props.defaultValue}, props)
+					}
+			}
+		case Constant.create:
+			return props
+	}
+}
+
+export function commit (element, name, value, xmlns, from) {
+	switch (name) {
+		case 'style':
+
+			break
+		case 'className':
+			if (!xmlns && value) {
+				return property(element, name, value)
+			}
+		case 'class':
+			return attribute(element, 'class', value, '')
+		case 'xlink:href':
+			return attribute()(element, name, value, 'http://www.w3.org/1999/xlink')
```

Comment:
?
should have failed both tests and coverage

Reply:
Not yet implemented. Tests haven't been ported yet.